web/static/js/LoginPage.jsx

```diff
@@ -0,0 +1,102 @@
+"use strict";
+
+var _ = require("lodash");
+var React = require("react");
+var utils = require('./utils');
+
+var LoginPage = React.createClass({
+  getInitialState: function() {
+    return {
+      username: '',
+      password: '',
+      error: '',
+      submitting: false
+    };
+  },
+
+  onUsernameChange: function(e) {
+    this.setState({ username: e.target.value });
+  },
+
+  onPasswordChange: function(e) {
+    this.setState({ password: e.target.value });
+  },
+
+  usernameIsValid: function() {
+    return this.state.username !== '';
+  },
+
+  passwordIsValid: function() {
+    return this.state.password !== '';
+  },
+
+  formIsValid: function() {
+    return _.every([
+        this.usernameIsValid(),
+        this.passwordIsValid()
+    ]);
```

Comment:
nit: can we just `this.usernameIsValid() && this.passwordIsValid()` here

Reply:
Copy/pasted from another component where validation is done.